src/main/java/org/isf/opd/service/OpdIoOperations.java

```diff
@@ -226,4 +231,51 @@ public Optional<Opd> getOpdById(Integer code) {
 	public List<Opd> getOpdByProgYear(Integer code) {
 		return repository.findByProgYear(code);
 	}
+	
+	public PagedResponse<Opd> getOpdListPageable(
+			Ward ward, 
+			DiseaseType diseaseType,
+			String diseaseCode,
+			LocalDateTime dateFrom,
+			LocalDateTime dateTo,
+			int ageFrom,
+			int ageTo,
+			char sex,
+			char newPatient,
+			String user,
+			int page,
+			int size) throws OHServiceException {
+		return setPaginationData(repository.findOpdListPageable(ward, diseaseType, diseaseCode, dateFrom, dateTo, ageFrom, ageTo, sex, newPatient, user, PageRequest.of(page, size)));
+	}
+	
+	/**
+	 * Return {@link List} of {@link Opd}s associated to specified patient ID
+	 * 
+	 * @param patID - the patient ID
+	 * @param page - page number
+	 * @param size - size of the list
+	 * @return the list of {@link Opd}s associated to specified patient ID.
+	 * 		   the whole list of {@link Opd}s if <code>0</code> is passed.
+	 * @throws OHServiceException 
+	 */
+	public PagedResponse<Opd> getOpdListPageables(int patID, int page, int size) throws OHServiceException {
+		return patID == 0 ?
+				setPaginationData(repository.findAllOrderByProgYearDescPageable(PageRequest.of(page, size))) :
+				setPaginationData(repository.findAllByPatient_CodeOrderByProgYearDescPageable(patID, PageRequest.of(page, size)));
+	}
+	
+	public PagedResponse<Opd> setPaginationData(Page<Opd> pages) {
```

Comment:
missing javadoc

Reply:
anyway is it not possible to factorize this method as `PagedResponse<T> setPaginationData(Page<T> pages)` ?